lexers/o/org.go

```diff
@@ -0,0 +1,80 @@
+package o
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Markdown lexer.
+var Org = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "Org Mode",
+		Aliases:   []string{"org", "orgmode"},
+		Filenames: []string{"*.org"},
+		MimeTypes: []string{"text/org"}, // https://lists.gnu.org/r/emacs-orgmode/2017-09/msg00087.html
+	},
+	Rules{
+		"root": {
+			{`^#\s.*$`, Comment, nil},
+			// Headings
+			{`^(\*+)( COMMENT)( .*)$`, ByGroups(GenericHeading, NameBuiltin, Text), nil},
```

Comment:
I was unable to have something like `` `^(\*+)( (COMMENT|DONE))( .*)$` `` working. Then we end up with 4 groups. So I changed the later part to `ByGroups(GenericHeading, Text, NameBuiltin, Text)`. But that did not work. Strangely that duplicated group 3 in the final output. So "* DONE foo" will show up as "* DONEDONE".

So what's the best way to optimize these six lines?

Reply:
Nested sub-groups need to be non-capturing (ie. `( (?:COMMENT|DONE))`).